src/python/pants/engine/target.py

```diff
@@ -351,9 +351,18 @@ def _calculate_field_values(
             valid_aliases.add(field_type.alias)
             aliases_to_field_types[field_type.alias] = field_type
             if field_type.deprecated_alias is not None:
+                valid_aliases.add(field_type.deprecated_alias)
                 aliases_to_field_types[field_type.deprecated_alias] = field_type
+
         for alias, value in unhydrated_values.items():
             if alias not in aliases_to_field_types:
+                if isinstance(self, TargetGenerator):
```

Comment:
Shouldn't this be handled in the previous loop, though? This will only augment the error message...

Reply:
That's right - we don't want to use the moved fields as actual values, only for the error message. I can improve the comment.